File: client/src/app/models/product.ts
export interface Product{
    id: number;
  name: string;
  description: string;
  price: number;
  pictureUrl: string;
  genre?: string;
  author: string;
  quantityInStock?: number;
}

export interface ProductParams {
  orderBy: string;
  searchTerm?: string;
  genres?: string[];
  authors?: string[];
  pageNumber : number;
  pageSize: number;
}

File: API/Entities/Product.cs
namespace API.Entities
{
    public class Product
    {
       
        public int Id { get; internal set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public long Price { get; set; }
        public string PictureUrl { get; set; }
        public string Genre { get; set; }
        public string Author { get; set; }
        public int QuantityInStock { get; set; }
    }
} 